autoload/codefmt/zprint.vim

```diff
@@ -26,6 +26,30 @@
 
 let s:plugin = maktaba#plugin#Get('codefmt')
 
+""
+" @private
+function! s:ApplyToZprintOptions(opts, key, value) abort
+  call maktaba#ensure#IsList(a:opts)
+  call maktaba#ensure#IsString(a:key)
+
+  " We can only apply Vim settings for simple values of a:opts, because it's
+  " an EDN map, not a list of command-line flags.
+  let l:edn = empty(a:opts) ? '{}' : a:opts[0]
+  if l:edn[0] != '{'
+    return a:opts
+  endif
+
+  " Don't try to insert if key already appears in the EDN.
+  if l:edn =~ a:key
```

Comment:
This pattern matching looks brittle. `=~` behavior depends on the `'ignorecase'` setting, and it'll match against values and parts of keys. Maybe you could split on whitespace and search through those strings, or something like that?

Reply:
Will address in a separate PR